src/utils/conversion.ts

```diff
@@ -3362,3 +3363,36 @@ export function agentGroupToPermissionGroup(
     }
   }
 }
+
+/**
+ * @hidden
+ */
+export async function portfolioNameToNumber(
+  rawIdentityId: IdentityId,
+  rawName: PortfolioName,
+  context: Context
+): Promise<BigNumber | null> {
+  const {
+    polymeshApi: {
+      query: { portfolio },
+    },
+  } = context;
+
+  const rawPortfolioNumber = await portfolio.nameToNumber(rawIdentityId, rawName);
+
+  const portfolioId = u64ToBigNumber(rawPortfolioNumber);
+
+  // TODO @prashantasdeveloper remove this logic once nameToNumber returns Option<PortfolioNumber>
+  if (portfolioId.eq(new BigNumber(1))) {
+    /**
+     * since nameToNumber returns for non-existing portfolios,
```

Comment:
```suggestion
     * since nameToNumber returns 1 for non-existing portfolios,
```

nit, but I think it makes a difference in the readability.